ui/packages/catalog-ui-search/src/main/webapp/react-component/utils/validation/validation.tsx

```diff
@@ -38,31 +52,344 @@ export function showErrorMessages(errors: any) {
-      })
-    }
-    if (
-      geometry &&
-      geometry.type === 'LineString' &&
-      geometry.coordinates.length < 2
-    ) {
-      errors.push({
-        title: 'Invalid geometry filter',
-        body: 'Line coordinates must be in the form [[x,y],[x,y], ... ]',
+    getGeometryErrors(filter).forEach(function(msg) {
+      geometryErrors.add(msg)
+    })
+  }
+  geometryErrors.forEach(function(err) {
+    errors.add({
+      title: 'Invalid geometry filter',
+      body: err,
+    })
+  })
+  return Array.from(errors)
+}
+
+export function validateGeo(
+  key: string,
+  value: string,
+  value1?: any,
+  value2?: any,
+  value3?: any
+) {
+  switch (key) {
+    case 'lat':
+      return validateDDLatLon(latitude, 90, value)
+    case 'lon':
+      return validateDDLatLon(longitude, 180, value)
+    case 'dmsLat':
+      return validateDmsLatLon(latitude, value)
+    case 'dmsLon':
+      return validateDmsLatLon(longitude, value)
+    case 'usng':
+      return validateUsng(value)
+    case 'utmUpsEasting':
+    case 'utmUpsNorthing':
+    case 'utmUpsZone':
+    case 'utmUpsHemisphere':
+      return validateUtmUps(key, value, value1, value2, value3)
+    case 'radius':
+    case 'lineWidth':
+      return validateRadiusLineBuffer(key, value)
+    case 'line':
+    case 'poly':
+    case 'polygon':
+      return validateLinePolygon(key, value)
+    default:
+  }
+}
+
+export function getErrorComponent(errorState: ErrorState) {
+  return errorState.error ? (
+    <Invalid>
+      <WarningIcon className="fa fa-warning" />
+      <span>{errorState.message}</span>
+    </Invalid>
+  ) : null
+}
+
+export function validateListOfPoints(coordinates: any[], mode: string) {
+  let message = ''
+  const isLine = mode.includes('line')
+  const numPoints = isLine ? 2 : 4
+  if (
+    !mode.includes('multi') &&
+    !coordinates.some(coords => coords.length > 2) &&
+    coordinates.length < numPoints
+  ) {
+    message = `Minimum of ${numPoints} points needed for ${
+      isLine ? 'Line' : 'Polygon'
+    }`
+  }
+  coordinates.forEach(coordinate => {
+    if (coordinate.length > 2) {
+      coordinate.forEach((coord: any) => {
+        if (hasPointError(coord))
+          message = JSON.stringify(coord) + ' is not a valid point.'
       })
+    } else {
+      if (mode.includes('multi')) {
+        message = `Switch to ${isLine ? 'Line' : 'Polygon'}`
+      } else if (hasPointError(coordinate)) {
+        message = JSON.stringify(coordinate) + ' is not a valid point.'
+      }
     }
+  })
+  return { error: message.length > 0, message }
+}
+
+function validateLinePolygon(mode: string, currentValue: string) {
+  if (!is2DArray(currentValue)) {
+    return { error: true, message: 'Not an acceptable value' }
+  }
+  try {
+    const pointsValid = validateListOfPoints(JSON.parse(currentValue), mode)
+    return pointsValid === undefined ? initialErrorState : pointsValid
+  } catch (e) {
+    return initialErrorState
```

Comment:
`is2DArray` is also doing a `JSON.parse` of the `currentValue`, and in the catch it returns false, causing this function to return `{ error: true, message: 'Not an acceptable value' }`. I think we should do the same thing here